src/frontend/mame/ui/state.cpp

```diff
@@ -0,0 +1,424 @@
+// license:BSD-3-Clause
+// copyright-holders:Nathan Woods
+/***************************************************************************
+
+    ui/state.cpp
+
+    Menus for saving and loading state
+
+***************************************************************************/
+
+#include "emu.h"
+#include "machine.h"
+#include "emuopts.h"
+#include "ui/state.h"
+
+
+/***************************************************************************
+    ANONYMOUS NAMESPACE
+***************************************************************************/
+
+namespace {
```

Comment:
This anonymous namespace should really be inside the `namespace ui` to avoid weird dependent name lookup issues.